pkg/operations/kubernetesupgrade/upgradecluster.go

```diff
@@ -101,10 +104,21 @@ func (uc *UpgradeCluster) UpgradeCluster(az armhelpers.AKSEngineClient, kubeConf
 		kubeClient = k
 	}
 
-	if err := uc.getClusterNodeStatus(kubeClient, uc.ResourceGroup); err != nil {
+	if err := uc.setNodesToUpgrade(kubeClient, uc.ResourceGroup); err != nil {
 		return uc.Translator.Errorf("Error while querying ARM for resources: %+v", err)
 	}
 
+	if kubeClient != nil {
```

Comment:
Do we need this nil check? We should be confident that a non-err response from L100 above will guarantee that `k` is not nil, right?

Reply:
If we're not confident then I'd just add that nil check to the err condition:

`if err != nil || k == nil {`

Reply:
the only reason is to pass test `Should not fail if a Kubernetes client cannot be created`, do not know the history so I am just complying